### PR TITLE
[5] Add code analysis rake task

### LINKED ISSUE #5: Add rake task for code analysis
### Why
We want to build top quality projects with static code analysis tools in place by default

### What
Add the rake task to run both Reek and Rubocop

### PR DESCRIPTION
**Pull request by juanmanuelramallo**
Resolves #5 

- Changed all File.open#read calls to just File.read
- Moved support files to the first level in the folder tree
- Added a task to copy over the code analysis script

### CODE REVIEW

File: lib/rootstrap/tasks/add_code_analysis.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        write_to_rakefile
+      end
+
+      private
+
+      def rakefile
+        @rakefile ||= File.read("#{File.dirname(__FILE__)}/../support/Rakefile")
+      end
+
+      def write_to_rakefile
```
Comment (fedeagripa) [inline lib/rootstrap/tasks/add_code_analysis.rb]: why is this private and you are accessing it like an alias method 🤔 ?

Comment (fedeagripa) [reply lib/rootstrap/tasks/add_code_analysis.rb]: If you really want to do an alias (or call methods different)I found this: https://stackoverflow.com/questions/31983479/alias-method-a-private-method that shows some options

Comment (juanmanuelramallo) [reply lib/rootstrap/tasks/add_code_analysis.rb]: Good point. I was just following the structure of the other tasks, but it's true that I can move the body of this method to the main one

File: lib/rootstrap/support/Rakefile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+task :code_analysis do
+  sh 'bundle exec rubocop lib spec'
+  sh 'bundle exec reek lib'
```
Comment (TimoPeraza) [inline lib/rootstrap/support/Rakefile:5]: don't we need to run ```rails_best_practices```?

Comment (juanmanuelramallo) [reply lib/rootstrap/support/Rakefile:5]: I don't think we do because the output project won't be a rails one but an empty gem

**Review by fedeagripa — APPROVED** [review]

**Review by TimoPeraza — APPROVED** [review]
🏅

**Pull request merged**